contracts/deploy/Create3.sol

```diff
@@ -2,45 +2,42 @@
 
 pragma solidity ^0.8.0;
 
+import { ICreate3 } from '../interfaces/ICreate3.sol';
 import { ContractAddress } from '../utils/ContractAddress.sol';
-import { CreateDeployer } from './CreateDeployer.sol';
-
-error AlreadyDeployed();
-error EmptyBytecode();
-error DeployFailed();
+import { Create } from './Create.sol';
 
 /**
- * @title Create3 Library
- * @notice This library can be used to deploy a contract with a deterministic address that only
- * depends on the sender and salt, not the contract bytecode.
+ * @title Create3 contract
+ * @notice This contract can be used to deploy a contract with a deterministic address that only
+ * depends on the sender and salt, not the contract bytecode and constructor parameters.
  */
-library Create3 {
+contract Create3 is ICreate3 {
     using ContractAddress for address;
 
     // slither-disable-next-line too-many-digits
-    bytes32 internal constant DEPLOYER_BYTECODE_HASH = keccak256(type(CreateDeployer).creationCode);
+    bytes32 internal constant DEPLOYER_BYTECODE_HASH = keccak256(type(Create).creationCode);
 
     /**
      * @notice Deploys a new contract using the CREATE3 method.
      * @dev This function first deploys the CreateDeployer contract using
-     * the CREATE2 opcode and then utilizes the CreateDeployerto deploy the
+     * the CREATE2 opcode and then utilizes the CreateDeployer to deploy the
      * new contract with the CREATE opcode.
      * @param salt A salt to further randomize the contract address
      * @param bytecode The bytecode of the contract to be deployed
      * @return deployed The address of the deployed contract
      */
-    function deploy(bytes32 salt, bytes memory bytecode) internal returns (address deployed) {
-        deployed = deployedAddress(address(this), salt);
+    function create3Deploy(bytes32 salt, bytes memory bytecode) internal returns (address deployed) {
```

Comment:
```suggestion
    function _deploy(bytes32 salt, bytes memory bytecode) internal returns (address deployed) {
```
maybe? create3 prefix seems redundant

Reply:
Had to leave create3 prefix. We already to the same thing with other utils like `safeTransfer` etc